workshops/3D Heatmap/README.md

```diff
@@ -0,0 +1,202 @@
+import pandas as pd
+import seaborn as sns
+```
+The first step is to create 3D Axes, and then plot any of the 3D graphs. We pass ```projection='3d'``` to plt.axes, This is the empty canvas that we'll be painting.
+
+```python
+fig = plt.figure()
+ax = plt.axes(projection='3d')
+plt.show()
+```
+![3D Plane](https://cloud-a08hob7s0.vercel.app/1workshop2.gif)
+
+# 2. Plotting Graph
+
+Plotting 3-D a Graph
+```python
+from mpl_toolkits.mplot3d import Axes3D
+import matplotlib.pyplot as plt
+import pandas as pd
+import seaborn as sns
+```
+The most basic three-dimensional plot is a line or collection of scatter plot created from sets of (x, y, z) triples.
+
+```
+ax = plt.axes(projection='3d')
+z = np.linspace(0, 30, 100)
+
+Initialise x is array of np.sin(z), and similarly y is a array of np.cos(z)
+```python
+x = np.sin(z)
+y = np.cos(z)
+```
+Shows  the graph
+```python
+ax.plot3D(x, y, z, 'red')
+plt.show()
+```
+
+![3D Graph](https://cloud-a08hob7s0.vercel.app/2workshop3.gif)
+
+
+# 3. Plotting with Dataset
+
+```python
+from mpl_toolkits.mplot3d import Axes3D
+import matplotlib.pyplot as plt
+import pandas as pd
+import seaborn as sns
+```
+
+Get the data (CSV file) from the file
+volcano [dataset](https://cloud-8ycpvzexa.vercel.app/0volcano.csv)
+
+```python
+data = pd.read_csv('volcano.csv')
+```
+
+Transform it to a long format to plot each data on the 3D plane
+```python
+df = data.unstack().reset_index()
+df.columns = ['X', 'Y', 'Z']
+```
+
+And transform the old column name into something numerical value. Plots have numerical values on the axis of the plane.
+```
+df['X'] = pd.Categorical(df['X'])
+df['X'] = df['X'].cat.codes
+```
+
+Make the plot in the 3D plane with the help of the projection function.
+Finally, trisurf creates a surface by first finding a set of triangles formed between adjacent points. 
+```
+fig = plt.figure()
+ax = fig.gca(projection='3d')
+ax.plot_trisurf(df['Y'], df['X'], df['Z'], cmap=plt.cm.jet, linewidth=0.2)
+```
+Add a color bar that maps values to colors. The point height on the mountain increase, the color value will shift from blue to red.
+
+```python
+surf = ax.plot_trisurf(df['Y'], df['X'], df['Z'], cmap=plt.cm.jet, linewidth=0.2)
+fig.colorbar(surf, shrink=0.5, aspect=5)
+```
+We will use this function to display the plot on our computer screen.
+```
+plt.show()
+```
+
+
+**Output**
+![Volcano Heatmap](https://cloud-94iqxy8lo.vercel.app/0volcano.gif)
+
+### Marvelous! You have created your first 3D Heat Map plot ###
+
+
+# voilà!
+
+
+**Congratulations!✌️ you have completed the workshop. Share it with your friends and family.
+Share with everyone🤗✌️🥳👏🏅🌇🎊**
+
+![congratulations g.i.f](https://cloud-1th3ydnib.vercel.app/2workshop_happy.gif)
+
+
+
+
+# Keep Hacking!
+Mountain Bruno [Code](https://repl.it/@ShiveshSingh/Mtbrunoplot)
+
+Surface Plot [Code](https://repl.it/@ShiveshSingh/Surface-Plot-3D#main.py)
+
+Heat Map Animation [Code](https://repl.it/@ShiveshSingh/HeatmapAnimation)
+
+
```

Comment:
My final thoughts

- Fix the second section called plotting a graph by importing numpy and by getting rid of the instructions in the code which causes syntax errors.
- Fix the continue hacking section or explain what I have to do to see it properly.
- Be a little more specific on complex line function especially with a complex library like pandas and the 3D mapping mostly in section 3

Reply:
Done